akka-actor-typed-tests/src/test/scala/akka/actor/typed/internal/routing/RoutingLogicSpec.scala

```diff
@@ -129,11 +132,55 @@ class RoutingLogicSpec extends ScalaTestWithActorTestKit with WordSpecLike with
 
       (0 to 10).foreach { _ =>
         // not much to verify here, but let's exercise it at least
-        val routee = logic.selectRoutee()
+        val routee = logic.selectRoutee(emptyMessage)
         routees should contain(routee)
       }
 
     }
 
   }
+
+  "The consistent hashing logic" must {
+    val behavior: Behavior[String] = Behaviors.empty[String]
+    val typedSystem: ActorSystem[String] = ActorSystem(behavior, "emptySystem")
+    val identityMapping: ConsistentHashMapping[String] = { case in: String => in }
+
+    "not accept virtualization factor lesser than 1" in {
+      val caught = intercept[IllegalArgumentException] {
+        new RoutingLogics.ConsistentHashingLogic[String](0, ConsistentHashingLogic.emptyHashMapping, typedSystem)
+      }
+      caught.getMessage shouldEqual "virtualNodesFactor must be >= 1"
+    }
+
+    "not accept null actor system" in {
+      val caught = intercept[IllegalArgumentException] {
+        new RoutingLogics.ConsistentHashingLogic[String](2, ConsistentHashingLogic.emptyHashMapping, null)
+      }
+      caught.getMessage shouldEqual "requirement failed: system argument of ConsistentHashingLogic cannot be null."
+    }
+
+    "return deadLetters when there are no routees" in {
+      val logic =
+        new RoutingLogics.ConsistentHashingLogic[String](1, identityMapping, typedSystem)
+      logic.selectRoutee("msg") shouldBe typedSystem.deadLetters
+    }
+
+    "hash consistently" in {
+
+      val refA = typedSystem.ref // -1670067195
```

Comment:
@patriknw, I think here I need your help I have following problem - when I create actors their `ActorRef`s are created in nondeterministic way - in order to have them unique, which makes sense.

In order to test consistent hashing - we need to have refs that are, well, consistent. The only way I was able to get consistent ref is by created whole `ActorSystem`. Is there any other possibility of doing that?

Reply:
That would be difficult. I think it's better to test that messages that hash to same value are routed to the same destination instead of testing that it should select a specific destination.